src/utils/jsx.ts

```diff
@@ -17,52 +6,22 @@ type MaybeArray<T> = T | T[]
  */
 type BaseAttributes = {
   style?: string | Record<string, string>
+  class?: string
   className?: string
   [key: string]: unknown
 }
 
-/**
- * Extracts properties from custom elements with strict typing.
- * Filters out methods and HTMLElement base properties.
- */
-type CustomElementProps<T extends HTMLElement> = Partial<{
-  [K in keyof T as T[K] extends (...args: never[]) => unknown ? never : K extends keyof HTMLElement ? never : K]: T[K]
-}> &
-  BaseAttributes
-
-/**
- * Extracts attributes from native HTML elements with flexible typing.
- * Allows string coercion for numeric and other properties (e.g., width="300").
- */
-type NativeElementProps<T extends HTMLElement> = Partial<{
-  [K in keyof T as K extends "style" ? never : K]: T[K] | string
+type ElementProps<T extends HTMLElement> = Partial<{
+  [K in keyof T as K extends "style" ? never : K]: T[K]
 }> &
   BaseAttributes
 
-/**
- * Maps all native HTML element tag names to their corresponding attribute types.
- * This provides proper type safety for standard HTML elements in JSX while maintaining flexibility.
- */
-type HTMLElementAttributes = {
-  [K in keyof HTMLElementTagNameMap]: NativeElementProps<HTMLElementTagNameMap[K]>
-}
-
 declare global {
   // eslint-disable-next-line @typescript-eslint/no-namespace
   namespace JSX {
     type Element = HTMLElement
-    interface IntrinsicElements extends HTMLElementAttributes {
-      "nosto-bundle": CustomElementProps<Bundle>
-      "nosto-campaign": CustomElementProps<Campaign>
-      "nosto-control": CustomElementProps<Control>
-      "nosto-dynamic-card": CustomElementProps<DynamicCard>
-      "nosto-image": CustomElementProps<Image>
-      "nosto-product": CustomElementProps<Product>
-      "nosto-section-campaign": CustomElementProps<SectionCampaign>
-      "nosto-simple-card": CustomElementProps<SimpleCard>
-      "nosto-sku-options": CustomElementProps<SkuOptions>
-      "nosto-variant-selector": CustomElementProps<VariantSelector>
-      "test-element": Record<string, unknown>
+    type IntrinsicElements = {
```

Comment:
since all custom element types are registered into HTMLElementTagNameMap it made sense to just remap the structure